File: app/models/product.py
from sqlalchemy import Column, BigInteger, Text, TIMESTAMP
from sqlalchemy.sql import func

from app.models.base import Base


class Product(Base):

    __tablename__ = "product"

    id = Column(BigInteger, primary_key=True)

    title = Column(Text, nullable=False)

    brand = Column(Text)

    category = Column(Text)

    source = Column(Text)

    created_at = Column(TIMESTAMP, server_default=func.now())

File: api-server/src/main/java/com/productresearch/apiserver/domain/product/entity/Product.java
package com.productresearch.apiserver.domain.product.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "product")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Builder
@AllArgsConstructor
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "source_product_id", unique = true)
    private Long sourceProductId;

    @Column(nullable = false)
    private String title;

    private String brand;
    private String category;
    private String source;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}
